controllers/storagecluster/uninstall_reconciler.go

```diff
@@ -325,9 +340,10 @@ func (r *StorageClusterReconciler) deleteResources(sc *ocsv1.StorageCluster) (re
 	}
 
 	objs := []resourceManager{
+		&storageClient{},
 		&ocsExternalResources{},
-		&ocsProviderServer{},
 		&ocsNoobaaSystem{},
+		&ocsProviderServer{},
```

Comment:
Why are we moving this after NooBaa? the order is, unfortunately, very delicate and as stated in a prev comment: the order should be reversed for uninstallation:
1. Noobaa
2. client 
3. API server

If not you will get into a dependency hell

Reply:
done.